docs/puml.sh

```diff
@@ -0,0 +1,19 @@
+#!/bin/bash
+set -eu
+if [[ ! -d docs ]] ; then
+  echo "execute from the root directory" && exit 1
+fi
+
+export PLANTUML_VERSION=1.2021.4
+
+function svg {
+  if [[ -f "docs/$1.puml" ]] ; then
+    echo "updating docs/$1.svg"
+    < "docs/$1.puml" docker run --rm -i karfau/plantuml:$PLANTUML_VERSION -pipe -tsvg -nometadata > "docs/$1.svg"
```

Comment:
For people like me who don't have so much extra horsepower to run Docker on my laptop, do you think it will work if I would use Ubuntu on the cloud? Would you recommend any resources if I need to install or configure anything?

Reply:
The alternatives to using docker are
- either have java installed and download the plantuml.jar and use the command `< "docs/$1.puml" java -jar path/to/plantuml.jar -pipe ...` instead. I added this as an option to the script, but depending on some OS dependencies (e.g. graphviz) versions the output can differ (this is what happened when I tried it locally).
- or use the web interface http://www.plantuml.com/plantuml/uml/SyfFKj2rKt3CoKnELR1Io4ZDoSa70000 (but I think it includes `metadata` which causes the generated to change every time)

Or we could of course also use some other format for creating images/diagrams for documentation purposes. This is just the one that supports may styles and I know how to use it.